### PR TITLE
[Feature] Feature Engineering

### PR DESCRIPTION
**Pull request by oo-ni**
## ✅ PR 유형
- [ ] Bugfix
- [x] Feature
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Code style update (formatting, local variables)
- [ ] Build
- [ ] Other...

## #️⃣ 연관된 이슈

> #14 

## 📝 작업 내용

> feature engineeing (train/test feature 전처리)
> 모델링용 feature 선택, label encoding, 정규화, 이상치 제거 등 수행
> 분류모델 활용(XGBoost) 고려해서 label encoding
> data leakage 방지용 is_train 로직 추가해서 train/test 분리

### 💬 리뷰 요구사항 (선택)

> 리뷰어가 특별히 봐주었으면 하는 부분이 있다면 작성해주세요
>
> ex) 메서드 XXX의 이름을 더 잘 짓고 싶은데 혹시 좋은 명칭이 있을까요?

### CODE REVIEW

File: transformer/ranking/feature_engineering.py [hunk 1]
```diff
@@ -0,0 +1,145 @@
+from typing import Tuple
+
+import pandas as pd
+import numpy as np
+from sklearn.preprocessing import StandardScaler, RobustScaler
+from sklearn.model_selection import train_test_split
+import joblib
+
+class FeatureEngineer:
```
Comment (sunnyineverywhere) [inline transformer/ranking/feature_engineering.py]: 클래스 이름만 `FeatureEngineer`에서 `FeatureFactory` 나 `FeatureService`같은 좀 더 용도가 명확학 이름으로 바꾸면 어떨까?
`FeatureEngineer`라는 이름만 보면 유저 이름 같은 느낌이 강해서 feature를 담당하는 클래스라고 인식되진 않는것같아

**Review by sunnyineverywhere — CHANGES_REQUESTED** [review]

**Review by sunnyineverywhere — APPROVED** [review]
